internal/services/mssqlmanagedinstance/mssql_managed_database_data_source.go

```diff
@@ -0,0 +1,192 @@
+	Name                    string                    `tfschema:"name"`
+	ResourceGroupName       string                    `tfschema:"resource_group_name"`
+	ManagedInstanceName     string                    `tfschema:"managed_instance_name"`
+	ManagedInstanceId       string                    `tfschema:"managed_instance_id"`
+	LongTermRetentionPolicy []LongTermRetentionPolicy `tfschema:"long_term_retention_policy"`
+	ShortTermRetentionDays  int64                     `tfschema:"short_term_retention_days"`
+	PointInTimeRestore      []PointInTimeRestore      `tfschema:"point_in_time_restore"`
+}
+
+var _ sdk.DataSource = MsSqlManagedDatabaseDataSource{}
+
+type MsSqlManagedDatabaseDataSource struct{}
+
+func (d MsSqlManagedDatabaseDataSource) ResourceType() string {
+	return "azurerm_mssql_managed_database"
+}
+
+func (d MsSqlManagedDatabaseDataSource) ModelObject() interface{} {
+	return &MsSqlManagedDatabaseDataSourceModel{}
+}
+
+func (d MsSqlManagedDatabaseDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.ManagedDatabaseID
+}
+
+func (d MsSqlManagedDatabaseDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ValidateMsSqlManagedInstanceDatabaseName,
+		},
+		"managed_instance_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ManagedInstanceID,
+		},
+	}
+}
+
+func (d MsSqlManagedDatabaseDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"managed_instance_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"resource_group_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"long_term_retention_policy": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					// WeeklyRetention - The weekly retention policy for an LTR backup in an ISO 8601 format.
+					"weekly_retention": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					// MonthlyRetention - The monthly retention policy for an LTR backup in an ISO 8601 format.
+					"monthly_retention": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					// YearlyRetention - The yearly retention policy for an LTR backup in an ISO 8601 format.
+					"yearly_retention": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					// WeekOfYear - The week of year to take the yearly backup in an ISO 8601 format.
+					"week_of_year": {
+						Type:     pluginsdk.TypeInt,
+						Computed: true,
+					},
+
+					"immutable_backups_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Computed: true,
+					},
+				},
+			},
+		},
+		// "long_term_retention_policy": helper.LongTermRetentionPolicySchema(),
```

Comment:
I'm not sure these comments are really all that helpful in the data source here
```suggestion
					"weekly_retention": {
						Type:     pluginsdk.TypeString,
						Computed: true,
					},

					"monthly_retention": {
						Type:     pluginsdk.TypeString,
						Computed: true,
					},

					"yearly_retention": {
						Type:     pluginsdk.TypeString,
						Computed: true,
					},

					"week_of_year": {
						Type:     pluginsdk.TypeInt,
						Computed: true,
					},

					"immutable_backups_enabled": {
						Type:     pluginsdk.TypeBool,
						Computed: true,
					},
				},
			},
		},
```